apps/gnome-web/src/store/useGnomeStore.ts

```diff
@@ -0,0 +1,59 @@
+import type { CreateGnomeRequest } from "@repo/shared/requests";
+import type { GnomeResponse } from "@repo/shared/responses";
+import { create } from "zustand";
+import { GnomesService } from "@/api/Gnomes.service";
+
+interface GnomeState {
+  gnomes: GnomeResponse[];
+  loading: boolean;
+  error: string | null;
+
+  fetchGnomes: () => Promise<void>;
+  addGnome: (payload: FormData) => Promise<GnomeResponse>;
+  removeGnome: (id: string) => Promise<void>;
+  updateGnome: (
+    id: string,
+    payload: Partial<CreateGnomeRequest>,
+  ) => Promise<GnomeResponse>;
+}
+
+export const useGnomeStore = create<GnomeState>((set) => ({
+  gnomes: [],
+  loading: false,
+  error: null,
+
+  fetchGnomes: async () => {
+    set({ loading: true, error: null });
+    try {
+      const data = await GnomesService.getGnomes();
+      set({ gnomes: data, loading: false });
+    } catch (error: any) {
+      if (error.response?.status === 401) {
+        set({ error: "Brak autoryzacji – zaloguj się", loading: false });
```

Comment:
zamiast robić takiego ifa w każdym zapytaniu do backendu da się to zrobić w samym axiosie, zerknij na coś takiego jak interceptory w axiosie